src/main/java/io/microprofile/config/Config.java

```diff
@@ -54,7 +50,7 @@
 	 * @param defaultValue the default value
 	 * @return the property value or the defult value if the property does not exist.
 	 */
-	<T> T getProperty(String propertyName, Class<T> propertyType, T defaultValue);
+	<T> T getPropertyValue(String propertyName, Class<T> propertyType, T defaultValue);
```

Comment:
I prefer Java7 as it means this config can be used in Java7 environment together with other EE7 technologies. On the other hands, Optional sounds appealing...